### PR TITLE
feat(events): Add payment metadata to hyperswitch-payment-intent-events

### PR DESCRIPTION
**Pull request by tsdk02**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

 - `metadata` is marked as `pii::SecretSerdeValue`, now changed it to `serde_json::Value` to remove the masking.
- Added the `metadata` field for `hyperswitch-payment-intent-events` (type - String).

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Adding the metadata to the payment-intent-event helps in providing better information about the event.

## How did you test it?

Make a normal payment through Postman by following the steps:
- Merchant Account - Create
- API Key - Create
- Payment Connector - Create
- Payments - Create

You should now be able to see the `metadata` field in hyperswitch-payment-intent-events when viewed in kafka-ui.
Screenshot 2024-07-01 at 6 57 29 PM
And can also be searched using global search in the dashboard.
Screenshot 2024-07-02 at 1 06 11 PM
Screenshot 2024-07-02 at 1 06 54 PM

Metadata is of type - String
Screenshot 2024-07-02 at 3 55 37 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by lsampras — APPROVED** [review]

**Review by Abhitator216 — APPROVED** [review]

**Comment by Narayanbhat166** [comment]
> metadata is marked as pii::SecretSerdeValue, now changed it to serde_json::Value to remove the masking

@tsdk02 why do we need to remove the masking for serde json value?

**Review by Narayanbhat166 — DISMISSED** [review]

**Comment by tsdk02** [comment]
> > metadata is marked as pii::SecretSerdeValue, now changed it to serde_json::Value to remove the masking
> 
> @tsdk02 why do we need to remove the masking for serde json value?

We wanted to use the metadata field for global_search as well for generating events, we're removing the constraint on metadata being pii sensitive since it is being sent out of the code for global search

**Review by lsampras — APPROVED** [review]

File: config/development.toml [hunk 1]
```diff
@@ -664,7 +664,7 @@ host = "https://localhost:9200"
 [opensearch.auth]
 auth = "basic"
 username = "admin"
-password = "admin"
```
Comment (Narayanbhat166) [inline config/development.toml:667]: can you revert this change?

File: docker-compose.yml [hunk 2]
```diff
@@ -382,6 +382,7 @@ services:
     environment:
       - "discovery.type=single-node"
       - OPENSEARCH_INITIAL_ADMIN_PASSWORD=0penS3arc#
+      - LOG_LEVEL=DEBUG
```
Comment (Narayanbhat166) [inline docker-compose.yml:385]: is this intended change here?

**Review by Narayanbhat166 — CHANGES_REQUESTED** [review]

File: config/development.toml (hunk 1 above)
Comment (tsdk02) [reply config/development.toml:667]: Actually this is a required change for running opensearch locally.

File: docker-compose.yml (hunk 2 above)
Comment (tsdk02) [reply docker-compose.yml:385]: Yes, as it helps in better logging.

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by lsampras — DISMISSED** [review]

File: crates/api_models/src/payments.rs [hunk 3]
```diff
@@ -453,7 +453,7 @@ pub struct PaymentsRequest {
 
     /// You can specify up to 50 keys, with key names up to 40 characters long and values up to 500 characters long. Metadata is useful for storing additional, structured information on an object.
     #[schema(value_type = Option<Object>, example = r#"{ "udf1": "some-value", "udf2": "some-value" }"#)]
-    pub metadata: Option<pii::SecretSerdeValue>,
+    pub metadata: Option<serde_json::Value>,
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:456]: We'd intentionally kept this a `Secret` since this was a field provided by the merchant, and to not expose possible sensitive information in application logs. If we're now removing the `Secret`, we may have to consider masking possible sensitive information to our best efforts, both in application logs and in ClickHouse events.

**Review by lsampras — APPROVED** [review]

File: crates/api_models/src/payments.rs (hunk 3 above)
Comment (lsampras) [reply crates/api_models/src/payments.rs:457]: adding an issue for this #5227 .
we need to add consistency for this,
either by masking/hashing/encrypting it everywhere and deciding on a proper way for this

**Review by lsampras — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
